opensoundscape/signal_processing.py

```diff
@@ -543,31 +545,44 @@ def gcc(x, y, cc_filter="phat", epsilon=0.001, radius=None):
     # order of outputs matches torch.correlate and scipy.signal.correlate
     cc = torch.concatenate((cc[-y.shape[0] + 1 :], cc[: x.shape[0]]))
 
+    # normalize the cross-correlation coefficients if using cc_norm
+    # this normalizes both by length and amplitude
```

Comment:
It's not clear to me why dividing by (norm(x)*norm(y)) will achieve normalization by length and amplitude. Can you clarify what normalization means and why this is the correct operation? 

Normalizing by length: I think this means a value "per unit time" 

Normalizing by amplitude: I think this means the maximum value will be 1

Reply:
after discussing, we realized that this formulation is actually normalizing by the RMS amplitude and for the duration (given an RMS amplitude). This means that if two signals are perfectly correlated and have means of 0, their cross-correlation score will be 1.0 after the normalization. 

I was confused by the division by `norm(x)*norm(y)`, but it helped me to think about it like this:
- normalizing each original signal by the rms (to achieve rms=1) would mean dividing by `sqrt(sum of squared values / n )` (n = number of samples)
- "normalizing" for duration, assuming an rms of 1, would mean dividing the cross-correlation (dot product) by n; or equivalently, dividing the original signals by `sqrt(n)`
- thus, combining these two normalization would require dividing the original signals by `sqrt(n) * sqrt(sum of squares) / sqrt(n)` which is `sqrt(sum of squares)`. 
- This is the same as doing the dot product of signal 1 and signal 2 first then dividing by `sqrt(sum of squares signal 1) * sqrt(sum of squares signal 2)`, ie `norm(sig1)*norm(sig2)` which is what is implemented here

Reply:
Thanks for noting down what we talked about. One other thing to note is that assuming a signal has zero-mean, the rms is the same as the standard deviation. For clarity of code, we should probably rewrite this as
```
# ensure  signals both have 0-mean
x = x - np.mean(x)
y = y - np.mean(y)

# normalize by magnitude of signal
x  = x / np.std(x)
y = y / np.std(y)

# normalize by signal length
x = x / np.sqrt(len(x))
y = y / np.sqrt(len(y))
```